File: src/main/java/com/optica/services/login/UserServiceImpl.java
package com.optica.services.login;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
import org.springframework.stereotype.Service;

import com.optica.entity.User;
import com.optica.repository.login.RoleRepository;
import com.optica.repository.login.UserRepository;

@Service
public class UserServiceImpl implements UserService {

	@Autowired
	private UserRepository userRepository;
	@Autowired
	private RoleRepository roleRepository;
	@Autowired
	private BCryptPasswordEncoder bCryptPasswordEncoder;

	@Override
	public void save(User user) {
		user.setPassword(bCryptPasswordEncoder.encode(user.getPassword()));
		userRepository.save(user);

	}

	@Override
	public User findByUsername(String username) {
		return userRepository.findFirstByUsername(username);
	}

	@Override
	public List<User> findAll() {
		return userRepository.findAll();
	}

	@Override
	public User findById(Long id) {
		return userRepository.findById(id).orElseThrow(() -> new IllegalStateException("User not found by id")); 
	}

}

File: src/main/kotlin/com/optica/services/login/UserServiceImpl.kt
package com.optica.services.login

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder
import org.springframework.stereotype.Service

import com.optica.entity.User
import com.optica.repository.login.RoleRepository
import com.optica.repository.login.UserRepository

@Service
class UserServiceImpl : UserService {

    @Autowired
    private var userRepository: UserRepository? = null
    @Autowired
    private val roleRepository: RoleRepository? = null
    @Autowired
    private val bCryptPasswordEncoder: BCryptPasswordEncoder? = null

    override fun save(user: User) {
        user.setPassword(bCryptPasswordEncoder!!.encode(user.password!!))
        userRepository!!.save(user)

    }

    override fun findByUsername(username: String?): User {
        return userRepository!!.findFirstByUsername(username)
    }

    override fun findAll(): List<User> {
        return userRepository!!.findAll()
    }

    override fun findById(id: Long?): User {
        return userRepository!!.findById(id!!).orElseThrow<IllegalStateException?> { IllegalStateException("User not found by id") }
    }

}
